corehq/apps/accounting/invoicing.py

```diff
@@ -529,15 +529,66 @@ def get_factory_by_feature_type(cls, feature_type):
 
     @property
     @memoized
-    def is_prorated(self):
+    def _subscription_ends_before_invoice(self):
+        return (
+            self.subscription.date_end
+            and self.subscription.date_end < self.invoice.date_end
+        )
+
+    @property
+    @memoized
```

Comment:
nit: `memoized` doesn't really do much for performance on these properties. If I tracked it down correctly, I believe that these are just properties on a model that's already loaded https://github.com/dimagi/commcare-hq/blob/0f814368e4984a8b348e1b759916feae73ff262a/corehq/apps/accounting/models.py#L1880 Its not clear to me that comparing 2 dates is a greater cost than looking up in a dictionary of results (and storing those results).

at worst, it could result in some weird bug if `date_start` could change in memory when this code is run because the `memoized` cache isn't cleared